ledger/src/blockstore.rs

```diff
@@ -2425,19 +2425,18 @@ impl Blockstore {
         }
 
         if !slot_entries.is_empty() {
-            let (mut data_shreds, mut coding_shreds) = shredder.entries_to_shreds(
+            for s in shredder.make_merkle_shreds_from_entries(
                 keypair,
                 &slot_entries,
                 is_full_slot,
                 chained_merkle_root,
-                0,    // next_shred_index
-                0,    // next_code_index
-                true, // merkle_variant
+                0, // next_shred_index
+                0, // next_code_index
                 &reed_solomon_cache,
                 &mut ProcessShredsStats::default(),
-            );
-            all_shreds.append(&mut data_shreds);
-            all_shreds.append(&mut coding_shreds);
+            ) {
+                all_shreds.push(s);
+            }
```

Comment:
nit: I guess this does avoid creating the intermediate `Vec`; I think we could accomplish the same but slightly cleaner IMO with
```rust
let shred_iter = shredder.make_merkle_shred_from_entries(...);
all_shreds.extend(shred_iter);
```